### PR TITLE
Add note for offloading behavior

### PR DESCRIPTION
**Pull request by Alexsandruss**

### CODE REVIEW

File: doc/sources/oneapi-gpu.rst [hunk 1]
```diff
@@ -107,3 +107,7 @@ An example on how to patch your code with Intel CPU/GPU optimizations:
                [8., 7.], [8., 8.], [25., 80.]], dtype=np.float32)
    with config_context(target_offload="gpu:0"):
       clustering = DBSCAN(eps=3, min_samples=2).fit(X)
+
+Note: current offloading behavior restricts fitting and inference of any model to be
+in the same context or absence of context. For example, model trained in context with
+target_offload="gpu:0" will throw error if inference is done outside of context.
```
Comment (napetrov) [inline doc/sources/oneapi-gpu.rst:113]:
```suggestion
Note: current offloading behavior restricts fitting and inference of any model to be
in the same context or absence of context. For example, model trained in GPU context with
target_offload="gpu:0" will throw error if inference is done in CPU context.
```

File: doc/sources/oneapi-gpu.rst [hunk 2]
```diff
@@ -107,3 +107,7 @@ An example on how to patch your code with Intel CPU/GPU optimizations:
                [8., 7.], [8., 8.], [25., 80.]], dtype=np.float32)
    with config_context(target_offload="gpu:0"):
       clustering = DBSCAN(eps=3, min_samples=2).fit(X)
+
+Note: current offloading behavior restricts fitting and inference of any model to be
```
Comment (alexandraepan) [inline doc/sources/oneapi-gpu.rst:111]:
```suggestion
Note: Current offloading behavior restricts fitting and inference of any models to be
```

File: doc/sources/oneapi-gpu.rst [hunk 3]
```diff
@@ -107,3 +107,7 @@ An example on how to patch your code with Intel CPU/GPU optimizations:
...
+in the same context or absence of context. For example, model trained in context with
```
Comment (alexandraepan) [inline doc/sources/oneapi-gpu.rst:112]:
```suggestion
in the same context or absence of context. For example, a model trained in the GPU context with
```

File: doc/sources/oneapi-gpu.rst (hunk 1 above)
Comment (alexandraepan) [inline doc/sources/oneapi-gpu.rst:113]:
```suggestion
target_offload="gpu:0" throws an error if the inference is made in the GPU context.
```

Comment (Alexsandruss) [reply doc/sources/oneapi-gpu.rst]: It's applicable for any context currently

Comment (Alexsandruss) [reply doc/sources/oneapi-gpu.rst]: Leaving of `outside of` changes meaning

**Review by alexandraepan — APPROVED** [review]

**Pull request merged**